platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/StateSignatureTransaction.java

```diff
@@ -84,50 +84,48 @@ public StateSignatureTransaction() {}
      */
     public StateSignatureTransaction(
             final long round,
-            @NonNull final Signature stateSignature,
-            @NonNull final Hash stateHash,
-            @Nullable final Hash epochHash) {
+            @NonNull final Bytes stateSignature,
+            @NonNull final Bytes stateHash,
+            @NonNull final Bytes epochHash) {
         this.round = round;
-        this.stateSignature = Objects.requireNonNull(stateSignature, "stateSignature must not be null");
-        this.stateHash = Objects.requireNonNull(stateHash, "stateHash must not be null");
-        this.epochHash = epochHash;
-    }
-
-    /**
-     * Same as {@link #StateSignatureTransaction(long, Signature, Hash, Hash)} but with epochHash set to null
-     */
-    public StateSignatureTransaction(
-            final long round, @NonNull final Signature stateSignature, @NonNull final Hash stateHash) {
-        this(round, stateSignature, stateHash, null);
+        this.stateSignature = new Signature(
+                SignatureType.RSA,
+                Objects.requireNonNull(stateSignature, "stateSignature must not be null")
+                        .toByteArray());
+        this.stateHash = new Hash(
+                Objects.requireNonNull(stateHash, "stateHash must not be null").toByteArray());
+        this.epochHash = epochHash.length() == 0 ? null : new Hash(epochHash.toByteArray());
     }
 
+    // The following 4 methods equate to the ones that will be generated by PBJ once we switch to StateSignaturePayload
     /**
      * @return the round number of the signed state that this transaction belongs to
      */
-    public long getRound() {
+    public long round() {
         return round;
     }
 
-    /**
-     * @return the signature on the state
-     */
-    public Signature getStateSignature() {
-        return stateSignature;
+    @NonNull
+    public Bytes signature() {
```

Comment:
Did you intentionally delete the javadoc for this method?

Reply:
accident, I put it back in now